### PR TITLE
Postgres: column name truncation spike

### PR DESCRIPTION
**Pull request by edgao**
stacked on https://github.com/airbytehq/airbyte/pull/36620. ~See also https://github.com/airbytehq/airbyte/pull/36808 for "just throw an error".~

* tried to just add a truncate thing
* naive catalogparser collision detection doesn't help here
  * it just enters an infinite loop
* add more logic to the collision solver to handle the truncation case

we end up with e.g. `ddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddddd` (63 chars long) and `ddddddddddddddddddddddddddddd461dddddddddddddddddddddd_rirqs2` (note the `461` in the middle), when the original column names are `d<510>d_rirqs1` and `d<510>d_rirqs2`. Not convinced this is actually great behavior.

also, fix the catalogparser tests to have more stringent asserts. Previously they were actually just completely useless, because we were mocking the wrong method. The asserts should now catch this problem, since we assert on the actual table/column names.

submitted https://github.com/airbytehq/airbyte-internal-issues/issues/6989 to make collision logic better in general.

### CODE REVIEW

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/typing_deduping/CatalogParser.kt [hunk 1]
```diff
@@ -161,6 +161,45 @@ constructor(
                 var i = 1
```
Comment (jbfbell) [inline airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/typing_deduping/CatalogParser.kt:161]: <i class='graphite__hidden'>[Re: line 104]<br/><br/></i>

wise suggestion we should probably follow<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36805?utm_source=unchanged-line-comment">Graphite</a>.</span>

**Review by jbfbell — COMMENTED** [review]
General thumbs up, just the `toStreamConfig` method is getting a little unruly and we should break it up into sub methods

**Review by jbfbell — APPROVED** [review]

File: airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresSQLNameTransformer.java [hunk 2]
```diff
@@ -8,6 +8,17 @@
 
 public class PostgresSQLNameTransformer extends StandardNameTransformer {
 
+  // I _think_ overriding these two methods is sufficient to apply the truncation logic everywhere
+  // but this interface + our superclass are weirdly complicated, so plausibly something is missing
+  @Override
+  public String getIdentifier(final String name) {
+    return truncate(super.getIdentifier(name));
```
Comment (gisripa) [inline airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresSQLNameTransformer.java:15]: I already made another [mess here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresSqlGenerator.java#L79) unify that with call to `getIdentifier` ?

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/typing_deduping/CatalogParser.kt (hunk 1 above)
Comment (edgao) [reply airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/typing_deduping/CatalogParser.kt:165]: done in https://github.com/airbytehq/airbyte/pull/36805/commits/c6101974d452aa98e2376068351d37689b32c6eb?w=1. also cleaned up some of the kotlin autoconvert cruft.

File: airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresSQLNameTransformer.java (hunk 2 above)
Comment (edgao) [reply airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresSQLNameTransformer.java:15]: done in https://github.com/airbytehq/airbyte/pull/36805/commits/8291ba5e1e4432e522e6ea70abfea830b3344fe3.

**Pull request merged**